### ISSUE TITLE
Can't build for iOS using carthage with Xcode 9

### ISSUE THREAD

**Issue by TheTrueTom** [core-plot/core-plot#348]
I'm using carthage 0.25.0 and trying to build branch release-2.3 from core-plot.

I have no problem building for macOS  with the command :

`carthage update core-plot --platform macOS`

However when trying for iOS with the command :

`carthage update core-plot --platform iOS`

I'm faced with the following error 

```
Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -workspace /Users/---/Carthage/Checkouts/core-plot/examples/CorePlotExamples.xcworkspace -scheme CorePlot\ iOS -configuration Release -derivedDataPath /Users/---/Library/Caches/org.carthage.CarthageKit/DerivedData/9.0_9A235/core-plot/c0aae0fedf1e9854b437568a6a63476f90bb7ee6 -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES build (launched in /Users/---/Carthage/Checkouts/core-plot)
```

Here is what the log has to report :

```
Build settings from command line:
    BITCODE_GENERATION_MODE = bitcode
    CARTHAGE = YES
    CODE_SIGN_IDENTITY = 
    CODE_SIGNING_REQUIRED = NO
    ONLY_ACTIVE_ARCH = NO
    SDKROOT = iphoneos11.0

note: Planning build
note: Using build description from disk
Build system information
error: unexpected duplicate task: CompileDTraceScript /Users/tom/Documents/Inibox/Programme/Inibox Power Plus/Carthage/Checkouts/core-plot/framework/TestResources/CorePlotProbes.d (in target 'CorePlot iOS')

Build system information
error: unexpected duplicate task: CompileDTraceScript /Users/tom/Documents/Inibox/Programme/Inibox Power Plus/Carthage/Checkouts/core-plot/framework/TestResources/CorePlotProbes.d (in target 'CorePlot iOS')

** BUILD FAILED **

```

I have also tried using `--no-use-binaries` without success.

Hope someone can help

Cheers

**Comment by acwright** [comment]
+1 Also does not build for Mac with Xcode 9.

Log:

/Users/acwrightdesign/Documents/Developer/XCode/CashierFuManager/Carthage/Checkouts/core-plot/framework/Source/CPTTradingRangePlot.m:1226:33: error: implicit conversion from nullable pointer 'NSNumber * _Nullable' to non-nullable pointer type 'ObjectType _Nonnull' (aka 'id') [-Werror,-Wnullable-to-nonnull-conversion]
    CPTNumberArray *yValues = @[[self cachedNumberForField:CPTTradingRangePlotFieldOpen recordIndex:idx],
                                ^
/Users/acwrightdesign/Documents/Developer/XCode/CashierFuManager/Carthage/Checkouts/core-plot/framework/Source/CPTTradingRangePlot.m:1227:33: error: implicit conversion from nullable pointer 'NSNumber * _Nullable' to non-nullable pointer type 'ObjectType _Nonnull' (aka 'id') [-Werror,-Wnullable-to-nonnull-conversion]
                                [self cachedNumberForField:CPTTradingRangePlotFieldClose recordIndex:idx],
                                ^
/Users/acwrightdesign/Documents/Developer/XCode/CashierFuManager/Carthage/Checkouts/core-plot/framework/Source/CPTTradingRangePlot.m:1228:33: error: implicit conversion from nullable pointer 'NSNumber * _Nullable' to non-nullable pointer type 'ObjectType _Nonnull' (aka 'id') [-Werror,-Wnullable-to-nonnull-conversion]
                                [self cachedNumberForField:CPTTradingRangePlotFieldHigh recordIndex:idx],
                                ^
/Users/acwrightdesign/Documents/Developer/XCode/CashierFuManager/Carthage/Checkouts/core-plot/framework/Source/CPTTradingRangePlot.m:1229:33: error: implicit conversion from nullable pointer 'NSNumber * _Nullable' to non-nullable pointer type 'ObjectType _Nonnull' (aka 'id') [-Werror,-Wnullable-to-nonnull-conversion]
                                [self cachedNumberForField:CPTTradingRangePlotFieldLow recordIndex:idx]];
                                ^
4 errors generated.

**Comment by TheTrueTom** [comment]
I actually didn't have a problem building for mac. I used the following command :

`carthage update core-plot --platform macOS`

cartFile is the following :

`github "core-plot/core-plot" "release-2.3"`

and Cartfile.resolved looks like this :

`github "core-plot/core-plot" "c0aae0fedf1e9854b437568a6a63476f90bb7ee6"`

Also I used the standard build system (in Xcode project settings).

Don't know if this will help you @eskroch but it seems weird that I can build it and @acwright can't.

**Comment by eskroch** [comment]
Using the release version of Xcode 9, I can build successfully for Mac but not iOS. I'm seeing the same duplicate task error reported above. I'll troubleshoot and hopefully fix it.

**Comment by eskroch** [comment]
This error is caused by the new Xcode 9 build system. Reverting to the "standard" build system fixed it. Fixed on the `release-2.3` branch.

**Comment by fbronner** [comment]
Still getting this issue on the release-2.3 branch with the new build.  When you said it was fixed on the release-2.3 branch, was it with the new build system?

**Comment by eskroch** [comment]
The new build system is causing the error. The fix was reverting to the old build system setting.

**Comment by mkubista** [comment]
Is there any chance to repair this, to be able to run xCode 9 builds with "New build system"? I have found out, that if I remove the code using CorePlotProbes defined functions, the project compiles just fine. But honestly, I didn't use probes in my life and I hope I understand, for what they are (only tracing so for debug purpose?).

**Comment by eskroch** [comment]
It's probably time to remove the probe. It's only for debugging and I don't think anyone uses it any more.

**Comment by eskroch** [comment]
Merged PR #360 on the `release-2.3` branch to remove the DTrace probe.
